docs/markdown/mhd_module.md

```diff
@@ -0,0 +1,128 @@
+# Magnetohydrodynamics Module
+
+The magnetohydrodynamics (MHD) module augments Quokka's Euler solver with
+magnetic fields while keeping the discrete divergence of $\vec{B}$ exactly
+zero. It is built on the same method-of-lines framework described in
+[Equations](equations.md), but introduces face- and edge-centred fields to
+support constrained transport.
+
+## Finite-volume discretization
+
+- Cell-centred conserved variables ($\rho$, $\rho\vec{v}$, $E_{\text{gas}}$ and
+  optional scalars) live in the same `MultiFab` as the hydrodynamics module.
+  Magnetic fluxes $B_1$, $B_2$, $B_3$ are stored on face-centred `MultiFab`s so
+  that each component is collocated with its normal face area.
+- Edge-centred electromotive forces (EMFs) are defined on a third staggered grid
+  and assembled on demand. This staggered arrangement lets the induction update
+  follow Stokes' theorem exactly, ensuring $\nabla \cdot \vec{B}=0$ to machine
+  precision on uniform grids.
+- Source terms (cooling, gravity, etc.) act on the cell-centred state. Since only
+  ideal MHD is supported, there are no source terms in the magnetic induction
+  equation.
+
+## Reconstruction and Riemann solves
+
+Each sweep through `computeHydroFluxes` performs the following steps for every
+coordinate direction:
+
+1. Convert conserved variables (plus face-centred magnetic fluxes) to primitive
+   variables with `HydroSystem::ConservedToPrimitive`.
+2. Build shock-flattening coefficients that detect strong compression and
+   locally limit the reconstruction stencil to maintain monotonicity.
+3. Reconstruct left/right interface states using the requested order:
+   donor-cell (1), piecewise linear with a minmod limiter (2), PPM (3) or
+   extrema-preserving PPM (5, the default). The transverse magnetic components
+   are reconstructed alongside the fluid primitives so that the Riemann solver
+   sees a consistent state.
+4. Solve the one-dimensional Riemann problem with the HLLD solver. The normal
+   magnetic field is taken directly from the face-centred data so that the
+   solution respects the jump condition $B_n^- = B_n^+$. The solver returns both
+   fluxes for the conserved state and a face-centred velocity that is reused by
+   particle advection and, optionally, by the EMF calculation.
+
+If the high-order solve produces negative densities, the code marks the
+affected cells and later recomputes their update with a first-order upwind flux
+(see *Fallbacks* below). Negative pressures are corrected separately by the
+dual-energy synchronisation step.
+
+## Edge-centred electromotive forces
+
```

Comment:
Not sure how you want to handle this, but these options are going to be changed very shortly (i.e., in a matter of days). I suppose it's no problem to put this in for now, but just be aware that it will be stale a week from now.

Reply:
Sure, that's fine with me. Happy to have it replaced in a near-future PR.